app/uk/gov/hmrc/individualsincomeapi/domain/v1/SaSubmission.scala

```diff
@@ -232,10 +234,38 @@ object SaIncomeSources {
   implicit val format: Format[SaIncomeSources] = Json.format[SaIncomeSources]
 }
 
+
+case class SourceAddress(line1: Option[String] = None,
+                         line2: Option[String] = None,
+                         line3: Option[String] = None,
+                         line4: Option[String] = None,
+                         line5: Option[String] = None,
+                         postcode: Option[String] = None,
+                         effectiveDate: Option[LocalDate] = None,
+                         addressType: Option[String] = None) {
+
+  def isEmpty: Boolean = this == SourceAddress()
+}
+
+object SourceAddress {
+  implicit val desReads: Reads[SourceAddress] = (
+    (__ \ "line1").readNullable[String] and
+      (__ \ "line2").readNullable[String] and
+      (__ \ "line3").readNullable[String] and
+      (__ \ "line4").readNullable[String] and
+      (__ \ "line5").readNullable[String] and
+      (__ \ "postalCode").readNullable[String] and
```

Comment:
This does not need to be overwritten. Can be changed to be a Format instead recucing code and complexity.